tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "ApiViewProcessor.hpp"
+#include "AstDumper.hpp"
+#include <cassert>
+#include <memory>
+#include <optional>
+#include <tuple>
+#include <vector>
+
+namespace clang {
+class Decl;
+class ASTContext;
+enum AccessSpecifier : int;
+} // namespace clang
+
+struct DumpNodeOptions
+{
+  bool DumpListInitializer{false};
+  bool NeedsLeftAlign{true};
+  bool NeedsLeadingNewline{true};
+  bool NeedsTrailingNewline{true};
+  bool NeedsTrailingSemi{true};
+  bool NeedsNamespaceAdjustment{true};
+  bool IncludeNamespace{false};
+  bool IncludeContainingClass{false};
+};
+
+struct AstNode
+{
+  virtual void DumpNode(AstDumper* dumper, DumpNodeOptions dumpOptions) = 0;
+  AstNode(clang::Decl const* decl);
+
+  virtual std::string_view const Namespace() { return ""; }
```

Comment:
Is my understanding correct that an AstNode with Namespace == "" would be in the global namespace?

Reply:
If so could you add that as a comment

Reply:
A global type is one whose `getName` matches its `getQualifiedNameAsString`. 

An AstNode does not have a name, nor does it have a namespace. Added comments to reflect that.